Servers/UI/OJS.Servers.Ui/ClientApp/src/hooks/common/use-internal-url-params.tsx

```diff
@@ -0,0 +1,51 @@
+import React, { createContext, useCallback, useContext, useMemo, useState } from 'react';
+
+import { IHaveChildrenProps } from '../../components/common/Props';
+
+interface IInternalUrlParamsContext {
+    state: {
+        params: any;
+    };
+    actions: {
+        setParams: (newParams: unknown) => void;
+    };
+}
+
+type IInternalUrlParamsProvider = IHaveChildrenProps
+
+const InternalUrlParamsContext = createContext<IInternalUrlParamsContext>({} as IInternalUrlParamsContext);
+
+const defaultState = { state: { params: {} as Record<string, unknown> } };
+
+const InternalUrlParamsProvider = ({ children }: IInternalUrlParamsProvider) => {
```

Comment:
`Internal` is not a good naming. Maybe `PathUrlParams`, `PlaceholderUrlParams`, `RouteUrlParams`?